### DISCUSSION TITLE
Should we keep `Query::get_component` and friends?

### DISCUSSION THREAD

**Discussion by nicopap** [bevyengine/bevy#9910] [Polls]
`Query` has methods to access individual components of the query for a specific entity.

But is it really useful? One can simply use `Query::get` and access the component they want.

Those methods is code we need to take care of. This is not only a risk in terms of soundness (since it touches fairly hairy parts of the ECS) but it's also a burden. For example, you need to implement a `get_component` method when implementing dynamic queries. We also need to document them and test them, add similar API to `QueryState`, add `component` (panicking version), etc.

If we remove them, we don't have to worry about any of what we mentioned above.

### Examinees

- [`Query::get_component`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#method.get_component)
- [`Query::get_component_mut`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#method.get_component_mut)
- [`Query::get_component_unchecked`](https://docs.rs/bevy/latest/bevy/ecs/prelude/struct.Query.html#method.get_component_unchecked)

### Alleged usefulness

- For generic APIs where you have some `Q: WorldQuery` and don't know what components it has/what position they might be in.

That's it? I've implemented a lot of bevy plugins and personally never came around a use case for `get_component`, but if you can give a compelling use-case I'll add it to the list.

**Comment by james7132** [comment]
A comment I made on Discord that is worth reposting here: More often than not, I've seen people just use Query::get and just discard the unused parts, and the compiler optimizes out the unused fetches. It's definitely less ergonomic, but the fetches in Query::get are 100% in alignment with our safety checks used in iteration.

**Comment by B-Reif** [comment]
I would imagine any use case for these methods would be better-served by dynamic queries.
